app/Domain/KeyVisibilityUpgrades/KeyVisibilityUpgrade.php

```diff
@@ -5,14 +5,29 @@
 namespace App\Domain\KeyVisibilityUpgrades;
 
 use App\Domain\Integrations\KeyVisibility;
+use DateTimeImmutable;
 use Ramsey\Uuid\UuidInterface;
 
-final readonly class KeyVisibilityUpgrade
+final class KeyVisibilityUpgrade
 {
+    private DateTimeImmutable $createdAt;
```

Comment:
Can't this be included as readonly in the constructor?
`createdAt` seems to be required (not nullable)

Reply:
I did hide it from the constructor, because when requesting an upgrade the `createdAt` is not needed, it is controlled by saving it inside the database.

Reply:
Than it makes sense to make `createdAt` `?DateTimeImmutable` right?

Otherwise you can do

```php
$upgrade = new KeyVisibilityUpgrade(Uuid::uuid4(), Uuid::uuid4(), KeyVisibility::all);
$upgrade->getCreatedAt(); // this won't return a DateTimeImmutable
```